python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -648,6 +649,70 @@ def get(self):
     assert ray.get(worker_2.get.remote()) == (2, "2.3.0")
 
 
+@contextmanager
+def chdir(dir):
+    old_dir = os.getcwd()
+    os.chdir(dir)
+    yield
+    os.chdir(old_dir)
+
+
+@pytest.mark.skipif(
+    os.environ.get("CI") and sys.platform != "linux",
+    reason="This test is only run on linux CI machines.")
+def test_runtime_env_override(call_ray_start):
+    # https://github.com/ray-project/ray/issues/16481
+
+    with tempfile.TemporaryDirectory() as tmpdir, chdir(tmpdir):
+        ray.init(address="auto", namespace="test")
+
+        @ray.remote
+        class Child:
+            def getcwd(self):
+                import os
+                return os.getcwd()
+
+            def read(self, path):
+                return open(path).read()
+
+            def ready(self):
+                pass
+
+        @ray.remote
+        class Parent:
+            def spawn_child(self, name, runtime_env):
+                child = Child.options(
+                    lifetime="detached", name=name,
+                    runtime_env=runtime_env).remote()
+                ray.get(child.ready.remote())
+
+        Parent.options(lifetime="detached", name="parent").remote()
+        ray.shutdown()
+
+        with open("hello", "w") as f:
+            f.write("world")
+
+        job_config = ray.job_config.JobConfig(runtime_env={"working_dir": "."})
+        ray.init(address="auto", namespace="test", job_config=job_config)
+
+        os.remove("hello")
+
+        parent = ray.get_actor("parent")
+
+        env = ray.get_runtime_context().runtime_env
+        del env["working_dir"]  # make sure to directly use the direcotry
+        print("Spawning with env:", env)
+        ray.get(parent.spawn_child.remote("child", env))
```

Comment:
I think another thing for the future is to automatically inherit the URIs from the job config into the runtime_env of the task, so that we don't have to manually get `ray.get_runtime_context().runtime_env` and pass it in to the task options.